junto-rust/zomes/core/code/src/expressions/channel.rs

```diff
@@ -0,0 +1,123 @@
+//Module to handle all channel related operations
+use hdk::{
+    error::ZomeApiResult,
+    error::ZomeApiError,
+    holochain_core_types::{
+        entry::Entry, 
+        cas::content::Address
+    }
+};
+
+use super::utils;
+use super::group;
+use super::user;
+use super::definitions::{
+    app_definitions,
+    function_definitions::{
+        FunctionDescriptor,
+        FunctionParameters
+    }
+};
+
+//Commits den entry to DHT and runs necassary hooks
+pub fn commit_den(entry: &Entry, user: &Address) -> Result<Address, String> {
```

Comment:
Might be useful to return a `ZomeApiResult` instead of a `Result` then you can use the `?` operator to tidy things up and ensure the errors are propagated correctly.